PerformanceCalculator/ProcessorWorkingBeatmap.cs

```diff
@@ -46,6 +48,30 @@ private static Beatmap readFromFile(string filename)
                 return Decoder.GetDecoder<Beatmap>(reader).Decode(reader);
         }
 
+        public static ProcessorWorkingBeatmap FromFileOrId(string fileOrId)
+        {
+            if (fileOrId.EndsWith(".osu"))
+            {
+                if (!File.Exists(fileOrId))
+                    throw new ArgumentException($"Beatmap file {fileOrId} does not exist.");
+
+                return new ProcessorWorkingBeatmap(fileOrId);
+            }
+
+            if (!int.TryParse(fileOrId, out var beatmapId))
+                throw new ArgumentException("Incorrect beatmap ID.");
```

Comment:
```suggestion
                throw new ArgumentException("Could not parse provided beatmap ID.");
```

minor but "incorrect" reads a bit weird when it's actually invalid.